src/create-extension-pack.js

```diff
@@ -99,6 +100,7 @@ const externalBuiltins = ['ms-vscode.references-view',
     fs.writeFileSync(licensePath, generateLicense());
     fs.writeFileSync(readmePath, generateReadme());
 
+    createYarnLock(packFolderPath);
```

Comment:
Here the `yarn.lock` file is created just before running `vsce`. Node filesystem API does not guaranty that, upon returning, what has been written by `writeFileSync()` has been flushed to disk by the operating system. Maybe `createYarnLock()` could use `readFileSync()` to confirm that the file exists before returning. Just checking for the file's existence is probably enough delay for it to be written :)

I have this related intermittent issue, where I believe that `yarn.lock` probably has not been written to disk yet (potentially also some of the changes to the files written just before `package.json`, `LICENSE.txt` and `README.md`): 

```
Adding: ms-vscode.references-view
Adding: ms-vscode.js-debug-companion
Adding: ms-vscode.js-debug
Generated package.json file at: /home/lmcmcds/marc/vscode-builtin-extensions/vscode-builtin-extensions/builtin-extension-pack-1.77.0/package.json
Output: info No lockfile found.
[1/5] Validating package.json...
[2/5] Resolving packages...
[3/5] Fetching packages...
[4/5] Linking dependencies...
[5/5] Building fresh packages...
success Saved lockfile.

node:internal/errors:857
  const err = new Error(message);
              ^

Error: Command failed: yarn list --prod --json
{"type":"error","data":"No lockfile in this directory. Run `yarn install` to generate one."}

    at ChildProcess.exithandler (node:child_process:402:12)
    at ChildProcess.emit (node:events:513:28)
    at maybeClose (node:internal/child_process:1100:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:304:5) {
  code: 1,
  killed: false,
  signal: null,
  cmd: 'yarn list --prod --json'
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```